src/helper/UserSession.cpp

```diff
@@ -68,12 +68,19 @@ namespace SDDM {
             if (env.value(QStringLiteral("XDG_SESSION_TYPE")) == QLatin1String("wayland") && env.value(QStringLiteral("XDG_SESSION_CLASS")) == QLatin1String("greeter")) {
                 m_wayland = new WaylandHelper(this);
                 m_wayland->setEnvironment(env);
-                if (!m_wayland->startCompositor(m_displayServerCmd))
+                connect(m_wayland, &WaylandHelper::failed, this, &UserSession::stop);
```

Comment:
Why do we use a different path for late failure compared to failing to start?

Reply:
But you're right that no-one seems to be calling stop()

Should this be called from: 
Backend::closeSession

?

Reply:
> Why do we use a different path for late failure compared to failing to start?
Which path are you referring to? ::setup calls exit() upon error

> Should this be called from `Backend::closeSession`?
This will have closeSession be called because `connect(m_session, &UserSession::finished, this, &HelperApp::sessionFinished);`

Reply:
I was referring to:
```
           if (!m_wayland->startCompositor(m_displayServerCmd)) {
                    Q_EMIT finished(1);
```

and asking why this isn't

                connect(m_wayland, &WaylandHelper::failed, this,[this]() {
                  Q_EMIT finished(1);
                });

Reply:
Addressed by having stop() emit finished.

I don't think it makes sense to stop when we failed to start.